### PR TITLE
ARROW-12643: [Governance] Added experimental repos guidelines.

### PR DESCRIPTION
**Pull request by jorgecarleitao**

### CODE REVIEW

File: docs/source/developers/experimental_repos.rst [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+-------
+
+* A committer *may* initiate experimental work by creating a separate git
+  repository within the Apache Arrow (e.g. via `selfserve <https://selfserve.apache.org/>`_)
+  and announcing it on the mailing list, together with its goals, and a link to the
+  newly created repository.
+* The committer *must* initiate an email thread with the sole purpose of
+  presenting updates to the community about the status of the repo.
+* There *must not* be official releases from the repository.
+* Merge / migrations / make official *must* be discussed over the mailing list.
```
Comment (andygrove) [inline docs/source/developers/experimental_repos.rst:41]:
```suggestion
* Any decision to make the experimental repo official in any way, whether by merging or migrating, *must* be discussed and voted on in the mailing list.
```

**Comment by jorgecarleitao** [comment]
@wesm @emkornfield @andygrove , is there anything we need to do here?

**Comment by emkornfield** [comment]
Sorry for the delay. LGTM to me.  I'll wait for feedback from others before merging.

**Review by andygrove — APPROVED** [review]
LGTM!

File: docs/source/developers/experimental_repos.rst [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+  including licensing checks.
+* The committer decides when the repository is archived.
+
+Repository management
+---------------------
+
+* The repository *must* be under `apache/`
+* The repository’s name *must* be prefixed by `arrow-experimental-`
+* The committer has full permissions over the repository (within possible in ASF)
+* Push / merge permissions *must only* be granted to Apache Arrow committers
```
Comment (andygrove) [inline docs/source/developers/experimental_repos.rst:52]: We should also require CI checks for ASF header in files

**Comment by jorgecarleitao** [comment]
if no-one opposes, I plan to merge this tomorrow.

**Pull request closed without merging**